src/arch/xtensa/include/arch/lib/wait.h

```diff
@@ -16,14 +16,10 @@
 
 #if (CONFIG_WAITI_DELAY)
 
-static inline void arch_wait_for_interrupt(int level)
+static inline void arch_wait_for_interrupt(void)
 {
 	int i;
 
-	/* need to make sure the interrupt level won't be lowered */
-	if (arch_interrupt_get_level() > level)
```

Comment:
Need to keep this panic as it's hard to debug when used incorrectly.

Reply:
Should he compare with 0 instead of level, given that `level` is no longer available?

Reply:
yes, we only permit this at level 0.

Reply:
@lgirdwood Not exactly, there are a couple of cases where it is called on a higher level and the caller is OK with lowering the level by waiti. I attached one such example in #2140, another one is LP wait. Do you prefer to keep existing duplicates and arch_wait_for_interrupt(5) calls just to pass this check?

Reply:
@mmaka1 my worry is that this wil be called from an incorrect level as part of normal runtime operations and not part of shut down. These were hard to debug last time, so it may be best to pass the level and provide some comments on safe usage ?

Reply:
@mmaka1 @lgirdwood We warp `asm volatile("waiti 0");` with this IRQ_LV on purpose.
In past, we do have waiti call during IRQ_LV2 and IRQ_LV5 and it can work in most case, but we will got random DSP panic with broken stack. So @lgirdwood suggest us do not call `asm volatile("waiti 0");` except we are in IRQ_LV0.

I am not sure why we want to go back to have the known issue again? Do we got any clarify from Xtensa that we can use it in other IRQ LV without issues?

Reply:
@xiulipan @lgirdwood I believe we agree that lowering to 0 is OK and `level` passed is just ignored while lowering the interrupt. So let's assume it should be kept and always set to 0 by a caller on xtensa platforms which would be consistent with the `waiti 0` and probably this should be tested (level == 0), not the against the current level which would make the caller to lower before calling into this function which would create ineffective transitions like 5 -> 0 outside / call / -> 5 -> inside/ 0 in some cases.

Reply:
@ktrzcinx Now the argument verification is consistent with the function behavior (level always set to 0 by waiti). However there are still calls in the code with level > 5 which will panic immediately.